advanced/matrix/matrix.py

```diff
@@ -0,0 +1,328 @@
+
+            ans_mat += (second_tup,)
+
+        return Matrix(ans_mat)
+
+    def opposite_matrix(self):
+        """
+        Changing the columns with the rows in Matrix object.
+        :param self: The Matrix object.
+        :return: A opposite Matrix. for example opposite_matrix((1,2,3),(4,5,6),(7,8,9)) = ((1,4,7),(2,5,8),(3,6,9))
+        """
+
+        opp_mat = ()
+        for index_in_tup in range(len(self._matrix)):
+            second_tup = ()
+
+            for tup_in_mat in range(len(self._matrix)):
+                second_tup += (self.tuples()[tup_in_mat][index_in_tup],)
+
+            opp_mat += (second_tup,)
+
+        return Matrix(opp_mat)
+
+    def __add__(self, other):
+        """
+        Method that implement a Matrix addition.
+        :param self: The first Matrix object.
+        :param other: The second Matrix object.
+        :return: New Matrix object which is the matrix sum result of self and other.
+        """
+        if type(other) != Matrix:
+            raise TypeError("Type error, can't add a '%s' to a Matrix objects" % other.__class__.__name__)
+
+        mat_result = ()
+        for i in range(len(self._matrix)):
+            mat_result += (tuple(map((lambda x, y: x + y), self.tuples()[i], other.tuples()[i])),)
+
+        return Matrix(mat_result)
+
+    def __sub__(self, other):
+        """
+        Method that implement a Matrix subtraction.
+        :param self: The first Matrix object.
+        :param other: The second Matrix object.
+        :return: New Matrix object which is the matrix subtraction result of other from self.
+        """
+        if type(other) != Matrix:
+            raise TypeError("Type error, can't subtract '%s' from Matrix objects" % other.__class__.__name__)
+
+        mat_result = ()
+        for i in range(len(self._matrix)):
+            mat_result += (tuple(map((lambda x, y: x - y), self.tuples()[i], other.tuples()[i])),)
+
+        return Matrix(mat_result)
+
+    def __truediv__(self, other):
+        """
+        Method that implement a Matrix division by scalar.
+        :param self: The first Matrix object.
+        :param other: Scalar (int or float object).
+        :return: New Matrix object which is the Matrix division by scalar.
+        """
+        if type(other) != int and type(other) != float:
+            raise TypeError("Type error raised, can't divide Matrix object by '%s' object." % other.__class__.__name__)
+
+        scalar = 1 / other
+        return self._scalar_multiplication(scalar)
+
+    def __eq__(self, other):
+        """
+        Method that implement a Matrix comparision.
+        :param self: The first Matrix object.
+        :param other: The second Matrix object.
+        :return: Boolean value - if the Matrix object are identical.
+        """
+        if type(other) != Matrix:
+            return False
+        return self.tuples() == other.tuples()
```

Comment:
If ill create 2 instances of a matrix with the same value, this function might return False, instead of True